src/library.js

```diff
@@ -824,12 +824,64 @@ LibraryManager.library = {
 #if SIMD
       196608
 #else
+#if WASM_ONLY
+      16384
+#else
       8192
 #endif
+#endif
     ) {
       return _emscripten_memcpy_big(dest|0, src|0, num|0)|0;
     }
 
+#if WASM_ONLY
+    ret = dest|0;
+    dest_end = (dest + num)|0;
+    if ((dest&7) == (src&7)) {
+      // The initial unaligned < 8-byte front.
+      while (dest & 7) {
+        if ((num|0) == 0) return ret|0;
+        store1(dest, load1(src, 1), 1);
+        dest = (dest+1)|0;
+        src = (src+1)|0;
+        num = (num-1)|0;
+      }
+      aligned_dest_end = (dest_end & -8)|0;
+      block_aligned_dest_end = (aligned_dest_end - 64)|0;
+      while ((dest|0) <= (block_aligned_dest_end|0) ) {
+        store8(dest,          load8(src,          8), 8);
+        store8(dest +  8 | 0, load8(src +  8 | 0, 8), 8);
+        store8(dest + 16 | 0, load8(src + 16 | 0, 8), 8);
+        store8(dest + 24 | 0, load8(src + 24 | 0, 8), 8);
+        store8(dest + 32 | 0, load8(src + 32 | 0, 8), 8);
+        store8(dest + 40 | 0, load8(src + 40 | 0, 8), 8);
+        store8(dest + 48 | 0, load8(src + 48 | 0, 8), 8);
+        store8(dest + 56 | 0, load8(src + 56 | 0, 8), 8);
+        dest = (dest+64)|0;
+        src = (src+64)|0;
+      }
+      while ((dest|0) < (aligned_dest_end|0) ) {
+        store8(dest, load8(src, 8), 8);
+        dest = (dest+8)|0;
+        src = (src+8)|0;
+      }
+    } else {
+      // In the unaligned copy case, unroll a bit as well.
+      aligned_dest_end = (dest_end - 4)|0;
+      while ((dest|0) < (aligned_dest_end|0) ) {
+        store4(dest, load4(src, 1), 1); // unaligned
```

Comment:
Ah right, that makes sense. Does this currently emit unaligned loads and stores on wasm or does it break up to 1 byte writes in binaryen?